### PR TITLE
feat: add Homepage layout

### PR DESCRIPTION
**Pull request by Seqlet**
https://www.figma.com/file/Dwjrkq9zwlhJxehVHgqrIg/UI-Main?node-id=164%3A510

### CODE REVIEW

File: src/modules/home/components/card-interest/CardInterest.vue
Comment (boompttn) [inline src/modules/home/components/card-interest/CardInterest.vue]:
```suggestion
    return { imgUrl, interestName, gradientColor, toggleSelect };

```

**Review by boompttn — APPROVED** [review]

File: src/modules/home/components/event-carousel/EventCarousel.vue
Comment (poom-kitti) [inline src/modules/home/components/event-carousel/EventCarousel.vue]: Naming should be "eventsList" to show plural

File: src/modules/home/components/event-carousel/EventCarousel.vue
Comment (poom-kitti) [inline src/modules/home/components/event-carousel/EventCarousel.vue]: Will it be easier to pass the whole event object as prop and let the card banner read the attribute of the event object itself instead of passing each individual attribute as prop? Not a suggestion to change but should be a convention we should discuss together,

File: src/modules/home/components/event-carousel/useEventCarousel.ts
Comment (new5558) [inline src/modules/home/components/event-carousel/useEventCarousel.ts]: Change any to interface

File: src/modules/home/components/event-carousel/EventCarousel.vue
Comment (new5558) [inline src/modules/home/components/event-carousel/EventCarousel.vue]: What is this?

File: src/modules/home/components/event-carousel/EventCarousel.vue
Comment (new5558) [inline src/modules/home/components/event-carousel/EventCarousel.vue:2]: Can we preview his component in storybook

File: src/modules/home/components/card-recommended/CardRecommended.vue
Comment (new5558) [inline src/modules/home/components/card-recommended/CardRecommended.vue]: type: Object as interfaceName

File: src/modules/home/Home.vue
Comment (new5558) [inline src/modules/home/Home.vue]: Why not use css class

File: src/modules/home/Home.vue
Comment (new5558) [inline src/modules/home/Home.vue]: Css class

Comment (new5558) [reply src/modules/home/Home.vue]: To clarify, css class load once, but style load twice.

File: src/modules/home/Home.vue
Comment (new5558) [inline src/modules/home/Home.vue]: Use class

File: src/modules/home/Home.vue
Comment (new5558) [inline src/modules/home/Home.vue]: Move this to new file. May name it useHome

File: src/modules/home/Home.vue
Comment (new5558) [inline src/modules/home/Home.vue:149]: Max width 961 and 960 is this intentional ?

File: src/modules/home/components/card-banner/CardBanner.vue
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.vue]: Why default to chula

File: src/modules/home/components/card-banner/CardBanner.vue
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.vue]: Why default to 99?

File: src/modules/home/components/card-banner/CardBanner.vue
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.vue]: Should we really have default , since this component wont load if there are no props in EvnerCarousel anyway

File: src/modules/home/components/card-banner/CardBanner.vue
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.vue]: Remove comments

File: src/modules/home/components/card-event/CardEvent.vue
Comment (new5558) [inline src/modules/home/components/card-event/CardEvent.vue]: create type for events prop

File: src/modules/home/components/card-interest/CardInterest.vue
Comment (new5558) [inline src/modules/home/components/card-interest/CardInterest.vue]: Why defualt cu

File: src/modules/home/components/card-organization/CardOrganization.vue
Comment (new5558) [inline src/modules/home/components/card-organization/CardOrganization.vue]: Add type

**Review by new5558 — CHANGES_REQUESTED** [review]

**Comment by new5558** [comment]
Since this is a big PR, please wait for design to approve this bf merge

File: src/modules/home/components/card-interest/CardInterest.vue
Comment (poom-kitti) [inline src/modules/home/components/card-interest/CardInterest.vue]: Do we still need this constant because we already have it as props?

**Comment by new5558** [comment]
4D5AC477-E053-498E-A580-2CDBEF0391C1
Isn’t design little bit strange on iPad? Especially “recommended for you”. Should the card be on center after it has expand to second row?

File: src/modules/home/Home.vue
Comment (poom-kitti) [inline src/modules/home/Home.vue]: Css class

File: src/modules/home/Home.vue
Comment (poom-kitti) [inline src/modules/home/Home.vue]: Css class

File: src/modules/home/Home.vue
Comment (poom-kitti) [inline src/modules/home/Home.vue]: Maybe we can refractor to file name testData.ts or something.

**Comment by poom-kitti** [comment]
When we hover over the "view more" word, the color of the right arrow icon should also change too. To implement it, pls refer to my back to top in PageFooter.vue.

**Comment by new5558** [comment]
image
บ่นๆๆๆๆๆๆ (test)

**Comment by new5558** [comment]
image
Please fix

**Comment by new5558** [comment]
image
Is text is a bit blurry?

File: src/modules/home/components/card-recommended/CardRecommended.vue
Comment (new5558) [inline src/modules/home/components/card-recommended/CardRecommended.vue]: I don't think we need default props, since the component will not be loaded anyway if there is no data.

File: src/modules/home/components/card-recommended/CardRecommended.vue
Comment (new5558) [inline src/modules/home/components/card-recommended/CardRecommended.vue]: delete comment

File: src/modules/home/components/card-recommended/CardRecommended.vue
Comment (new5558) [inline src/modules/home/components/card-recommended/CardRecommended.vue]: `recommended.placeholder`

Also, add `placeholder` in recommended props.

File: src/modules/home/components/card-banner/CardBanner.stories.ts
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.stories.ts:37]: add `imagePlaceholder` for blurhash

File: src/modules/home/components/card-banner/CardBanner.vue
Comment (new5558) [inline src/modules/home/components/card-banner/CardBanner.vue]: use computed

File: src/modules/home/components/card-event/CardEvent.stories.ts
Comment (new5558) [inline src/modules/home/components/card-event/CardEvent.stories.ts:31]: **blurhash** as props

**Review by new5558 — CHANGES_REQUESTED** [review]

**Review by new5558 — APPROVED** [review]
LGTM after fixed

**Comment by new5558** [comment]
Also, where is the Storybook for `eventCarousel` component
image

**Comment by new5558** [comment]
Please fix this on ipad safari 14.4
2460043D-8E4C-4627-A3EC-88759AAA5AF8

**Comment by new5558** [comment]
closed due to #18

**Pull request closed without merging**
